lemur/plugins/lemur_gcp/certificates.py

```diff
@@ -110,6 +111,31 @@ def get_self_link(project, name):
     return f"https://www.googleapis.com/compute/v1/projects/{project}/global/sslCertificates/{name}"
 
 
+def find_cert(project_id, credentials, body, cert_self_links):
+    """
+    Fetches the certificate bodies for each self_link and returns the first match by body.
+    :param project_id:
+    :param credentials:
+    :param body:
+    :param cert_self_links:
+    :return: The self link with a matching body, if it exists.
+    """
+    client = ssl_certificates.SslCertificatesClient(credentials=credentials)
+    for self_link in cert_self_links:
+        name = utils.get_name_from_self_link(self_link)
+        cert_meta = client.get(project=project_id, ssl_certificate=name)
+        # This should never happen.
+        if not cert_meta:
+            raise Exception(f"could not find metadata for certificate {name}")
```

Comment:
It seems this check can be removed?

Reply:
Between the time that we fetch ssl certs attached to the endpoint and when we fetch metadata for those, we could've detached and deleted it manually. It's unlikely to happen, but this also makes it explicit

Reply:
I'm surprised the SDK would return an empty object in that case, I thought it raises an exception if the status code is not found.

Reply:
ack, it does. I removed the check